mono/utils/mono-error.c

```diff
@@ -945,9 +800,15 @@ mono_error_set_from_boxed (MonoError *oerror, const MonoErrorBoxed *box)
 	DUP_STR (full_message);
 	DUP_STR (full_message_with_fields);
 	DUP_STR (first_argument);
-	DUP_STR (member_signature);
 	to->exn.klass = from->exn.klass;
 		  
 #undef DUP_STR
 	return (to->flags & MONO_ERROR_INCOMPLETE) == 0 ;
 }
+
+void
+mono_error_set_first_argument (MonoError *oerror, const char *first_argument)
+{
+	MonoErrorInternal* to = (MonoErrorInternal*)oerror;
+	to->first_argument = g_strdup (first_argument);
```

Comment:
I think this will leak`first_argument` unless you set the `MONO_ERROR_FREE_STRINGS` flag.